test/message/test_message.py

```diff
@@ -1,9 +1,107 @@
 import syft as sy
+import torch as th
 
+from syft import messaging
 
-def test_message_serde(hook):
 
-    x = sy.Message(0, [1, 2, 3])
+def test_message_serde():
+
+    x = messaging.Message(0, [1, 2, 3])
+    x_bin = sy.serde.serialize(x)
+    y = sy.serde.deserialize(x_bin, sy.local_worker)
+
+    assert x.contents == y.contents
+    assert x.msg_type == y.msg_type
+
+
+def test_cmd_message(workers):
+
+    bob = workers["bob"]
+
+    x = th.tensor([1, 2, 3, 4]).send(bob)
+
+    y = x + x  # this is the test
+
+    y = y.get()
+
+    assert (y == th.tensor([2, 4, 6, 8])).all()
+
+
+def test_obj_message(workers):
+
+    bob = workers["bob"]
+
+    x = th.tensor([1, 2, 3, 4]).send(bob)  # this is the test
+
+    y = x + x
+
+    y = y.get()
+
+    assert (y == th.tensor([2, 4, 6, 8])).all()
+
+
+def test_obj_req_message(workers):
+
+    bob = workers["bob"]
+
+    x = th.tensor([1, 2, 3, 4]).send(bob)
+
+    y = x + x
+
+    y = y.get()  # this is the test
+
+    assert (y == th.tensor([2, 4, 6, 8])).all()
+
+
+def test_get_shape_message(workers):
+
+    bob = workers["bob"]
+
+    x = th.tensor([1, 2, 3, 4]).send(bob)
+
+    y = x + x
+
+    z = y.shape  # this is the test
+
+    assert z == th.Size([4])
+
+
+def test_force_object_delete_message(workers):
+
+    bob = workers["bob"]
+
+    x = th.tensor([1, 2, 3, 4]).send(bob)
+
+    id_on_worker = x.id_at_location
+
+    assert id_on_worker in bob._objects
+
+    del x
+
+    assert id_on_worker not in bob._objects
+
+
+def test_is_none_message(workers):
+
+    bob = workers["bob"]
+
+    x = th.tensor([1, 2, 3, 4]).send(bob)
+
+    y = th.tensor([1]).send(bob)
+    y.child.id_at_location = x.id_at_location
+
+    assert not bob.request_is_remote_tensor_none(x)
+    assert not x.child.is_none()
+
+    del x
+
+    assert y.child.is_none()
+
+
+def test_search_message_serde():
```

Comment:
Can you add a test where these are each used in practice to make sure the messages are the right type? ie when deleting a tensor making sure a obj_delete_msg is made

Reply:
Yeah - it'll take me some time though

Reply:
All good. Take your time

Reply:
done